#! /bin/sh /usr/share/dpatch/dpatch-run
## 03_fix_tcp_wrappers.dpatch by Daniel Kahn Gillmor <dkg@fifthhorseman.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix calls to tcp wrappers

@DPATCH@
diff -urNad redir-2.2.1~/redir.c redir-2.2.1/redir.c
--- redir-2.2.1~/redir.c	2005-10-22 22:10:11.439455392 -0400
+++ redir-2.2.1/redir.c	2005-10-22 22:10:51.625346208 -0400
@@ -802,8 +802,8 @@
 #ifdef USE_TCP_WRAPPERS
 	request_init(&request, RQ_DAEMON, ident, RQ_FILE, clisock, 0);
 	sock_host(&request);
-	sock_hostname(&request);
-	sock_hostaddr(&request);
+	sock_hostname(request.client);
+	sock_hostaddr(request.client);
 
 	if (!hosts_access(&request)) {
 		refuse(&request);
@@ -1057,8 +1057,8 @@
 #ifdef USE_TCP_WRAPPERS
 		request_init(&request, RQ_DAEMON, ident, RQ_FILE, 0, 0);
 		sock_host(&request);
-		sock_hostname(&request);
-		sock_hostaddr(&request);
+		sock_hostname(request.client);
+		sock_hostaddr(request.client);
 	
 		if (!hosts_access(&request))
 			refuse(&request);
